pkg/sources/github/github.go

```diff
@@ -259,24 +324,66 @@ func (s *Source) visibilityOf(repoURL string) (visibility source_metadatapb.Visi
 	return
 }
 
+// Chunks emits chunks of bytes over a channel.
+func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) error {
+	apiEndpoint := s.conn.Endpoint
+	if len(apiEndpoint) == 0 || endsWithGithub.MatchString(apiEndpoint) {
+		apiEndpoint = "https://api.github.com"
+	}
+
+	installationClient, err := s.enumerate(ctx, apiEndpoint)
+	if err != nil {
+		return err
+	}
+
+	return s.scan(ctx, installationClient, chunksChan)
+}
+
+func (s *Source) enumerate(ctx context.Context, apiEndpoint string) (*github.Client, error) {
+	var installationClient *github.Client
+	var err error
+
+	switch cred := s.conn.GetCredential().(type) {
+	case *sourcespb.GitHub_Unauthenticated:
+		s.enumerateUnauthenticated(ctx)
```

Comment:
Ah yea, it looks like it's because all the other cases return the error. `enumerateUnauthenticated` doesn't return an error (it actually returns void) which is why this one doesn't have a return.